auth-service/src/main/kotlin/uk/gov/communities/delta/auth/utils/EmailAddressChecker.kt

```diff
@@ -7,16 +7,13 @@ import java.util.*
 
 class EmailAddressChecker {
 
-    fun hasValidEmailFormat(email: String): Boolean {
+    fun hasValidFormat(email: String): Boolean {
         if (email.count { it == '@' } != 1) return false
-
-        val emailRegex = """^[\w-+.']+@([\w-']+\.)+[\w-]{2,4}$""".toRegex()
         val cn = email.replace("@", "!")
-        return emailRegex.matches(email) && LDAPConfig.VALID_USERNAME_REGEX.matches(cn)
+        return LDAPConfig.VALID_EMAIL_REGEX.matches(email) && LDAPConfig.VALID_USERNAME_REGEX.matches(cn)
```

Comment:
Really minor points, feel free to ignore me:

Is it even possible to match the email regex and not the username regex?

Also, not escaping the hyphen feels slightly risky - it possibly works currently because it's currently next to `\w`, but it can be a special character inside `[]`, denoting a range of characters, like `a-z`

Reply:
I don't think it is possible to match email but not username but it is possible to match username but not email, but possibly both is useful incase a requirement changes and is updated in one place but not the other (although they are now in the same file so that is harder to do)

Reply:
There's a user called "admin" on staging which is shared by some DLUHC folks for testing. And there's our various test users in the test environment. But the normal sign up process isn't possible for those anyway so I don't think we need to worry about allowing it.

Reply:
`ADLdapLoginService` is also used for doing service user logins (like delta.app), so don't want to make it more restrictive. I think it's probably fine to check both

Reply:
(resolving conversation, will rename it next time that stuff is changed)